src/OpenTelemetry/BaseExportProcessor.cs

```diff
@@ -47,6 +48,16 @@ public sealed override void OnStart(T data)
 
         public override void OnEnd(T data)
         {
+            var activity = data as Activity;
```

Comment:
this will add unwanted perf impact, when using LogRecord.
I think we need to add BaseActivityExportProcessor, which inherits `BaseExportProcessor<Activity>`.

Reply:
It will be same as `SimpleActivityExportProcessor` and `BatchActivityExportProcessor` right?
The reason we wanted to put it here is to cover cases where exporters are inheriting from BaseExportProcessor and not overriding `OnEnd`.

Instead of introducing new one - should we add this check in Batch and Simple Activity processor (this is what I had before).